drivers/hmis/app/graphql/mutations/refresh_external_submissions.rb

```diff
@@ -0,0 +1,20 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Mutations
+  class RefreshExternalSubmissions < CleanBaseMutation
+    field :success, Boolean, null: false
+
+    def resolve
+      handlers = ['HmisExternalApis::ConsumeExternalFormSubmissionsJob']
+      return if Delayed::Job.queued?(handlers) || Delayed::Job.running?(handlers)
+
+      queue = ENV.fetch('DJ_SHORT_QUEUE_NAME', :short_running)
+      HmisExternalApis::ConsumeExternalFormSubmissionsJob.delay(priority: 12, queue: queue)
+
+      { success: true }
```

Comment:
I think the ux we’re proposing is that a user hits ‘refresh’, the button spins for a second, then that query resolves, but no new records are added to the table, because a delayed job was kicked off. Then, a few seconds/minutes later, the user would need to refresh the page to actually see the new records.

I'm wondering whether this query should *also* accept a form definition ID arg and return the latest results? That way at least the user would be able to hit refresh over and over and eventually see some new data. Maybe I misunderstood the plan for the UI of this feature, though - let's discuss more.

Reply:
Another option would be for the frontend to invalidate the cached submissions when this mutation returns successfully, to trigger a re-fetch of the table. That way, if the job _did_ get queued and executed immediately, they would see those results. It's still not ideal but might be an easy way to improve the experience slightly.

Reply:
Sounds like with the long-running queue it would take a few minutes for it to get queued, so it probably _won't_ be ready immediately. That said, if they click the button over and over, eventually the cache-clearing will yield results. So I think we should go with that.